model_utilities/train/finetuner.py

```diff
@@ -42,53 +44,68 @@ def __plot_losses(self, training_losses: list[float], validation_losses: list[fl
         plt.ylabel("Loss")
         plt.legend()
         plt.savefig("./debug/graphs/losses.png")
-
-    def __train(self, model: AutoModelForSeq2SeqLM, tokenizer: NllbTokenizer, dataset: ds.Dataset, optimizer: Adafactor,
+        
+    def __tokenize(self, dataset: ds.Dataset, src_lang: str, tgt_lang: str, tokenizer: NllbTokenizerFast, max_length : int) -> BatchEncoding:
+        tokenizer.src_lang = src_lang
+        tokenizer.tgt_lang = tgt_lang
+        tokenized_input = tokenizer(dataset[src_lang], text_target=dataset[tgt_lang], return_tensors='pt', padding=True, truncation=True, max_length=max_length)
+        tokenized_input.labels[tokenized_input.labels == tokenizer.pad_token_id] = -100
+        return tokenized_input
+    
+    def __tokenize_and_cleanup_dataset_pairs(self, dataset: ds.Dataset, tokenizer: NllbTokenizerFast, max_length : int) -> tuple[Dataset, Dataset]:
```

Comment:
This function violates the single responsibility principle. Please refactor by creating a `__normalize_dataset` function that takes `tokenized_dataset`, `tokenizer`, and `max_length` as parameters. Create `tokenized_train_dataset_pl_to_csb` and `tokenized_train_dataset_csb_to_pl` outside of this function, then pass them as arguments to `__normalize_dataset`.